internal/controller/snapshot/snapshot_adapter.go

```diff
@@ -729,3 +837,79 @@ func (a *Adapter) updateComponentSource(ctx context.Context, c client.Client, co
 	}
 	return nil
 }
+
+func (a *Adapter) prepareGroupSnapshot(application *applicationapiv1alpha1.Application, prGroupHash string) (*applicationapiv1alpha1.Snapshot, []gitops.ComponentSnapshotInfo, error) {
+	applicationComponents, err := a.loader.GetAllApplicationComponents(a.context, a.client, application)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	snapshotComponents := make([]applicationapiv1alpha1.SnapshotComponent, 0)
+	componentSnapshotInfos := make([]gitops.ComponentSnapshotInfo, 0)
+	for _, applicationComponent := range *applicationComponents {
```

Comment:
Similar to above, the group Snapshot should always contain all components of the application even if they weren't updated as part of the PR group - in that case they would just reference the GCL images. 
I think this loop is missing the logic to add those components to the Snapshot.

Reply:
Updated, thanks.